### PR TITLE
minor typos fixes, a potential optional addition,  and some more example usage

### PR DESCRIPTION
**Pull request by salguodnz**
Hey kazuhiro,

Firstly, I love this framework. it makes many many many gross terrible things much nicer to read and write through the builder-style pattern. 

In this PR, I have corrected a couple of typo's, so that it is easier to discover this framework's functionality.

I also have added an optional extension - which makes it easier to use an optional String with StringStylizer.

Let me know what you think!

### CODE REVIEW

File: StringStylizer/String+StringStylizer.swift [hunk 1]
```diff
@@ -47,3 +47,11 @@ public extension String {
...
 }
 
+
+public extension Optional where Wrapped == String {
+    
+    func stylize() -> StringStylizer<Styling> {
+        return (self ?? "").stylize()
+    }
+    
+}
```
Comment (kazuhiro4949) [inline StringStylizer/String+StringStylizer.swift:57]: Thank you for your suggestion. I'd like to know a use case of this method. When a variable is optional, you can unwrap it as follows (named Optional Chaining).

```swift
let str: String? = "foo"

str?.stylize() // => Optional<StringStylizer<Styling>>
```

**Comment by kazuhiro4949** [comment]
@salguodnz Thank you for your suggestion and fixing some typos :)

File: StringStylizer/String+StringStylizer.swift (hunk 1 above)
Comment (salguodnz) [reply StringStylizer/String+StringStylizer.swift:57]: yeah, I was aware of this - but it gets a little messy when you're combining optional `NSAttributedString`'s with non optional ones via the `stylize()` api. I had defined this extension in a codebase I'm working on, and will show you a with vs. without comparison.

 I've really enjoyed writing code with `StringStylizer` in the form of
```swift
attributedString = (
 someString.stylizer().<stylecode>.attr +
someOtherString.stylizer().<stylecode>.attr
)
```
I think this approach reads really well.


for example - consider this situation.

```swift
//for some reason, this `Product` has strange optionality - as in, productName may be nil.
        titleLabel.attributedText = (
            "\(product.productQuantity)x ".stylize().font(.preferredFont(forTextStyle: .headline)).attr +
            product.productName.stylize().font(.preferredFont(forTextStyle: .callout)).attr
        )
```

and compare that to this.

```swift
        titleLabel.attributedText = (
            "\(product.productQuantity)x ".stylize().font(.preferredFont(forTextStyle: .headline)).attr +
                (product.productName?.stylize().font(.preferredFont(forTextStyle: .callout)).attr ?? NSAttributedString())
        )
```

I think the first option reads better as it is more communicating the intent, not as awkwardly dealing with optionals to please the compiler :)

But I am fine with this extension being more specific to my use-case - and I am happy to remove this change if you deem it unsuited for the rest of your Framework consumers.

**Comment by salguodnz** [comment]
I've got another bit of feedback to do with optionals as well - whether you feel this is something you'd want to cater for in the current StringStylizer DSL or not - thats another question - but what would be awesome is if I could conditionally apply a stylizer. This is slightly different to the optional String suggestion in this PR.

For example: 
Here is a bunch of some of the more _convoluted_ string formation we have.

```swift
  let descAttrStrBuilder = promotion.dealDescription.stylize()
            .color(.normalContentTitle())
            .font(.preferredFont(forTextStyle: .subheadline))
        
        if promotion.isPromotionActionable {
            descAttrStrBuilder.underline(.single)
        }
        
        let descAttrStr = descAttrStrBuilder.attr
        
        var validUntil: String?
        if let endsDate = promotion.offerEndsDate {
            let endsDateString = DateFormatter.localizedString(from: endsDate, dateStyle: .short, timeStyle: .none)
            validUntil = "\nOffer Ends: \(endsDateString)"
        }
        
        let endDateAttrStr = validUntil.stylize()
            .color(.normalContentSubTitle())
            .font(.preferredFont(forTextStyle: .footnote))
            .attr
        
        titleLabel.attributedText = descAttrStr + endDateAttrStr
```

for the `promotion.isPromotionActionable` block - I wonder if this is something that could be made a 'first class' citizen of the current StringStylizer DSL.

What I mean by this is, It would be cool to write that if block 'in-line' like one of the below two syntaxes.

1.
```swift
descAttrStrBuilder.conditionally(promotion.isPromotionActionable).underline(.single)
```
 
OR

2.
```swift
descAttrStrBuilder.underline(.single, conditionally: promotion.isPromotionActionable)
```


Option 1 -  would mean adding more state into StringStylizer - when a `conditionally()` is used, you would need to evaluate that in the next 'style code' call (i.e. `underline()`) and then reset it.
The only issue I have with this is if there is more than 1 'style code' method called. i.e. `stylizer.conditionally(true).underline(.single).color(black)`
Does the conditional apply to both `underline()` and `color()`? 
That question is something that this syntax couldn't really communicate unless you were to change conditionally to `conditionallySingle()` and `conditionallyAll()` or something like this. It still isn't necessarily natural to consume though.

Option 2 - would be probably easier to understand as a framework consumer - and perhaps could be optional and default to `nil` or `true` :). It is not quite as pretty looking as option 1 however.

Either way would involve a pretty decent rewrite. I would be happy to help out if you like such a suggestion. 
Or alternatively, of course, you could suggest a better way to write the above snippet 😬

File: StringStylizer/String+StringStylizer.swift (hunk 1 above)
Comment (kazuhiro4949) [reply StringStylizer/String+StringStylizer.swift:57]: Thanks! That makes sense. I will merge this method to the new version. Can you make some unit tests in Test Target?

Comment (salguodnz) [reply StringStylizer/String+StringStylizer.swift:57]: Of course! I'll update some of the documentation (both in the Readme.md and in the code docs).

Comment (salguodnz) [reply StringStylizer/String+StringStylizer.swift:57]: @kazuhiro4949 , sorry for the long time to update. Please have a look through and let me know if you'd like me to change anything!

**Comment by kazuhiro4949** [comment]
@salguodnz Thanks!

**Pull request merged**